web-common/src/features/sources/modal/file-upload.ts

```diff
@@ -26,7 +23,7 @@ import {
  */
 export async function* uploadTableFiles(
   files: Array<File>,
-  [models, sources]: [Array<string>, Array<string>],
+  allNames: Array<string>,
   instanceId: string,
   goToIfSuccessful = true,
 ): AsyncGenerator<{ tableName: string; filePath: string }> {
```

Comment:
**Scalability Issue**: The change from specific model and source name checks to a generic allNames check in uploadTableFiles function simplifies the code but may inadvertently increase the risk of handling large datasets inefficiently. <br> **Fix**: Reintroduce more granular checks or implement a caching strategy for name validation to ensure scalability. <br> **Code Suggestion**: 
 ```
 -  allNames: Array<string>,
 +  [models, sources]: [Array<string>, Array<string>],
 ```

Reply:
Agree, I think this should only use Source names.